client/src/project/StateSelection.vue

```diff
@@ -0,0 +1,114 @@
+
+<script>
+  let blurTimeout;
+
+  export default {
+    props: ['value', 'hasError'],
+    data() {
+      return {
+        selected: null,
+        other: null,
+        states: [
+          { value: 'AL', text: 'Alabama' },
+          { value: 'AK', text: 'Alaska' },
+          { value: 'AZ', text: 'Arizona' },
+          { value: 'AR', text: 'Arkansas' },
+          { value: 'CA', text: 'California' },
+          { value: 'CO', text: 'Colorado' },
+          { value: 'CT', text: 'Connecticut' },
+          { value: 'DE', text: 'Delaware' },
+          { value: 'DC', text: 'District Of Columbia' },
+          { value: 'FL', text: 'Florida' },
+          { value: 'GA', text: 'Georgia' },
+          { value: 'HI', text: 'Hawaii' },
+          { value: 'ID', text: 'Idaho' },
+          { value: 'IL', text: 'Illinois' },
+          { value: 'IN', text: 'Indiana' },
+          { value: 'IA', text: 'Iowa' },
+          { value: 'KS', text: 'Kansas' },
+          { value: 'KY', text: 'Kentucky' },
+          { value: 'LA', text: 'Louisiana' },
+          { value: 'ME', text: 'Maine' },
+          { value: 'MD', text: 'Maryland' },
+          { value: 'MA', text: 'Massachusetts' },
+          { value: 'MI', text: 'Michigan' },
+          { value: 'MN', text: 'Minnesota' },
+          { value: 'MS', text: 'Mississippi' },
+          { value: 'MO', text: 'Missouri' },
+          { value: 'MT', text: 'Montana' },
+          { value: 'NE', text: 'Nebraska' },
+          { value: 'NV', text: 'Nevada' },
+          { value: 'NH', text: 'New Hampshire' },
+          { value: 'NJ', text: 'New Jersey' },
+          { value: 'NM', text: 'New Mexico' },
+          { value: 'NY', text: 'New York' },
+          { value: 'NC', text: 'North Carolina' },
+          { value: 'ND', text: 'North Dakota' },
+          { value: 'OH', text: 'Ohio' },
+          { value: 'OK', text: 'Oklahoma' },
+          { value: 'OR', text: 'Oregon' },
+          { value: 'PA', text: 'Pennsylvania' },
+          { value: 'RI', text: 'Rhode Island' },
+          { value: 'SC', text: 'South Carolina' },
+          { value: 'SD', text: 'South Dakota' },
+          { value: 'TN', text: 'Tennessee' },
+          { value: 'TX', text: 'Texas' },
+          { value: 'UT', text: 'Utah' },
+          { value: 'VT', text: 'Vermont' },
+          { value: 'VA', text: 'Virginia' },
+          { value: 'WA', text: 'Washington' },
+          { value: 'WV', text: 'West Virginia' },
+          { value: 'WI', text: 'Wisconsin' },
+          { value: 'WY', text: 'Wyoming' },
+          { value: 'other', text: 'Other/Non-USA' },
+        ],
+      };
+    },
+    computed: {
+      state() {
+        return this.selected === 'other' ? this.other : this.selected;
+      },
+    },
+    watch: {
+      state(val) {
+        this.$emit('input', val);
+      },
+      value() {
+        this.updateValue();
+      },
+    },
+    methods: {
+      onBlur() {
+        blurTimeout = window.setTimeout(() => {
```

Comment:
Why the setTimeout ?

Reply:
Woudln't Vue.NextTick be better?

Reply:
The setTimeout is because I don't want to trigger a validation error if you're just moving from the select field to the input field. By using setTimeout, I can clear the timeout on focus so the blur doesn't trigger. I use the same method in vue-dob-picker and have never had any issues with it.

With Vue.nextTick I'd have to do something like set a boolean onFocus and check that and reset it inside the nextTick callback. Also not sure if nextTick will delay things enough (although it certainly should).